tests/kafkatest/services/security/security_config.py

```diff
@@ -102,10 +128,11 @@ def __init__(self, security_protocol=None, interbroker_security_protocol=None,
         self.properties = {
             'security.protocol' : security_protocol,
             'ssl.keystore.location' : SecurityConfig.KEYSTORE_PATH,
-            'ssl.keystore.password' : SecurityConfig.ssl_stores['ssl.keystore.password'],
-            'ssl.key.password' : SecurityConfig.ssl_stores['ssl.key.password'],
+            'ssl.keystore.password' : SecurityConfig.ssl_stores.keystore_passwd,
+            'ssl.key.password' : SecurityConfig.ssl_stores.key_passwd,
             'ssl.truststore.location' : SecurityConfig.TRUSTSTORE_PATH,
-            'ssl.truststore.password' : SecurityConfig.ssl_stores['ssl.truststore.password'],
+            'ssl.truststore.password' : SecurityConfig.ssl_stores.truststore_passwd,
+            'ssl.endpoint.identification.algorithm' : 'HTTPS',
```

Comment:
@ijuma These properties are used by clients and enable hostname verification in clients. The broker uses properties in `tests/kafkatest/services/kafka/templates/kafka.properties` and that has been modified to enable hostname verification as well,